// d4-temperature-sensor/d4-temperature-sensor.js

import { LitElement, html, css } from "lit";

export class SensorCounterElement extends LitElement {

    static properties = {
        _counterState: {state: true},
        _temperatureState: {state: true},
        _intervalRef: {state: true},
        longitude: {},
        latitude: {}
    };

    constructor() {
        super();
        this._counterState = 0;   
    }

    getWeatherData() {
        const url = `https://api.open-meteo.com/v1/forecast?latitude=${this.latitude}&longitude=${this.longitude}&hourly=temperature_2m,relativehumidity_2m,windspeed_10m&current_weather=true`;
        fetch(url).then(response => response.json()).then(response => {
            const temperature = response.current_weather.temperature
            this._temperatureState = temperature
        }).catch(err => {
            console.error('Error:', err);
        })
    }
    
    connectedCallback() {
        super.connectedCallback();
        this.getWeatherData();

        this._intervalRef = setInterval(() => {
            this.getWeatherData();
        }, 3600000)
    }

    disconnectedCallback() {
        super.disconnectedCallback();
        // this.stopCounter();
        clearInterval(this._intervalRef);
    }


    startCounter() {
        this._intervalRef = setInterval(() => {
            this.incrementCounter();
        }, 1000)
    }

    stopCounter() {
        clearInterval(this._intervalRef);
    }

    incrementCounter() {
        this._counterState++;
    }

    static get styles() {
        return css`
            .container {
                background-color: #EEEEEE;
                height: 100px;
                width: 100px;
                display: flex;
                justify-content: center;
                align-content: center;
                align-items: center;
                border-radius: 15px;
                font-size: 2em;
                padding: 30px;
            }
        `
    }

    render() {
        return html`
            <!-- <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Zen+Script"> -->
            <div class="container">
                ${this._temperatureState}°C
            </div>   
        `
    }


}


window.customElements.define('d4-temperature-sensor', SensorCounterElement)
